modules/mongodb/mongodb.go

```diff
@@ -122,16 +105,66 @@ func (c *MongoDBContainer) ConnectionString(ctx context.Context) (string, error)
 	if err != nil {
 		return "", err
 	}
+
+	connStr := strings.Builder{}
+	connStr.WriteString("mongodb://")
+
 	if c.username != "" && c.password != "" {
+		connStr.WriteString(fmt.Sprintf("%s:%s@", c.username, c.password))
 		return fmt.Sprintf("mongodb://%s:%s@%s:%s", c.username, c.password, host, port.Port()), nil
 	}
 
-	endpoint, err := c.Endpoint(ctx, "mongodb")
+	connStr.WriteString(fmt.Sprintf("%s:%s", host, port.Port()))
+
+	if c.replicaSet != "" {
+		connStr.WriteString(fmt.Sprintf("/test?replicaSet=%s&directConnection=true", c.replicaSet))
+	}
+
+	return connStr.String(), nil
+}
+
+func replicaSetLifecycleHooks() testcontainers.ContainerLifecycleHooks {
+	return testcontainers.ContainerLifecycleHooks{
+		PostStarts: []testcontainers.ContainerHook{
```

Comment:
I think it would be clearer if we append the container hook functions directly to the PostStarts hook, instead of appending a complete lifecycle hook struct. In the end, the result is the same, but having them all together in the same lifecycle hook seems better aligned, as they all belong to the same hook's post-start events. Wdyt?

Reply:
I move `replicaSetPostStart` code into `replicaSetLifecycleHooks`, right? @mdelapenya

Reply:
You could leverage the `Options` pattern we are seen in other modules, such as Redpanda, where the Options struct holds all the values to be applied and all the functional options to configure the module are setting the values into that options struct.

Finally, I'd define a container lifecycle hook in that options, and with each call to the `WithReplicaSet` function, I'd be appending to the PostStarts field of the lifecycle hook. Last thing would be to append the mongo lifecycle hook to the container request before starting the container, right after all the options have been applied.

Does it make sense to you?

Reply:
@trungdlp-wolffun please let us know if you need anything from us 🙏

Reply:
@mdelapenya Thanks for your help, I will back soon.